src/tests/src/libs/antares/study/thermal-price-definition/thermal-price-definition.cpp

```diff
@@ -185,15 +175,14 @@ BOOST_AUTO_TEST_CASE(EconomicInputData_loadFromFolder_working_with_too_much_valu
 
 BOOST_AUTO_TEST_CASE(checkFuelAndCo2_basic_working)
 {
-    createIniFile();
     area->thermal.list.loadFromFolder(*study, folder, area);
```

Comment:
About `area->thermal.list.loadFromFolder(*study, folder, area);` : 

A test should have one purpose. We suppose here that the current test's purpose is to compare the number of TS of economical and technical sides. 

Why do we load a cluster from a file if we want to test the consistency of technical and economical TS sizes of the cluster ? We can simply create a cluster with the characteristics that we want.

```C++
   ThermalCluster cluster(area);
   
   // Technical characteristics of the cluster
   cluster.series = new DataSeriesCommon;
   cluster.series ->timeSeries.reset(1, 8760);

   // Economical characteristics of the cluster
   cluster.ecoInput.fuelcost.width = 1;
   cluster.ecoInput.co2cost.width = 1;

   area->thermal.list.mapping["my cluster"] = &cluster;
 
   BOOST_CHECK_NO_THROW(Antares::Check::checkFuelAndCo2ColumnNumber(study->areas));
```
Note that all loading functions disappear, including use of class **FuelCostFile** and **CO2CostFile**

Reply:
Yes; I did something extremely close. I used `Matrix<>::reset` instead of setting the `width` manually.